website/blog/2023-03-07-becoming-openid-certified-why-standards-matter/index.md

```diff
@@ -0,0 +1,66 @@
+
+The exact standard definition is linked, which makes it very easy to dig in further and figure out what's supposed to happen.
+
+For authentik, we use these tests to ensure that we adhere to the standard as much as possible to increase compatibility as much as possible. We've got all tests passing on the current development builds (which actually helped us a couple of bugs), and we'll be certifying the identity provider portion of authentik very soon.
+
+![Successful tests](./summary-green.png)
+
+### Deviation of standards
+
+It is sadly also quite common for clients to deviate from these standards, often requiring special behaviour that is specific to one or more implementations. One example that we had this issue with in the past is VMware's vCenter.
+
+They introduced "Identity federation" in [Version 7](https://blogs.vmware.com/vsphere/2020/04/vsphere-7-new-generation-vsphere.html), but explicitly only supported Microsoft's ADFS identity provider. However under the hood, this was using OIDC, which ADFS has supported for a while.
+
+Of course for us, ADFS being the only supported solution wasn't great, and I was very curious to see if ADFS was _actually_ required. With the OIDC implementation authentik had at the time, we didn't get very far. Logins would fail with a very cryptic error message, and since this was unsupported territory, there wasn't much we could find to help us figure out what's wrong.
+
+I'll spare you the details, but after a lot of digging through logs and figuring out what vCenter actually expects and attempts to do, I found it expects the `access_token` to be an encoded [JWT](https://jwt.io/) (stay tuned for our upcoming blog post about how JWT took the identity world by storm). Expecting an encoded JWT is not part of the [OpenID standard](https://openid.net/specs/openid-connect-core-1_0.html), so it somewhat made sense that they only advertise ADFS compatibility. However as we were finding out, vCenter was not the only applications that had this requirement. Researching further, it seemed like this had become sort of a "quasi-standard", as many identity providers were behaving this way.
+
+In the end we decided to follow suit with authentik (mostly for the sake of compatibility, but also since it can make sense), and vCenter logins via authentik are now [fully supported](https://goauthentik.io/integrations/services/vmware-vcenter/) (at least from our side).
+
+### Standards in authentik
+
+Quickly touching on standards more generally in authentik; we aim to make authentik as standards-compliant as possible while retaining its feature set. For example, for SAML sources/providers, all generated responses are tested against the official SAML XML schema. The same is done for the newly added SCIM integration, where everything is equally validated.
```

Comment:
```suggestion
Quickly touching on standards more generally in authentik; we aim to make authentik as standards-compliant as possible while retaining its feature set. For example, for SAML sources/providers, all generated responses are tested against the official SAML XML schema. The same is done for the newly added [SCIM](https://en.wikipedia.org/wiki/System_for_Cross-domain_Identity_Management) integration, where user identity information is equally validated across all systems and domains.
```

Reply:
Oh yeah that is pretty unclear; yeah I wanted to say to stay as close as possible to OIDC standards (or rather standards in general) while also retaining our feature set

I think for this post it makes more sense to be OIDC-compliant since that's the topic, but the general goal is to be compliant with all standards authentik supports